Blish HUD/GameServices/ContentService.cs

```diff
@@ -101,21 +107,39 @@ public enum FontStyle {
         }
 
         public Microsoft.Xna.Framework.Content.ContentManager ContentManager => Blish_HUD.BlishHud.ActiveContentManager;
-        
+
         protected override void Initialize() { /* NOOP */ }
 
         protected override void Load() {
             Textures.Load();
+            var zipArchiveReader = new ZipArchiveReader(RefPath);
+            _audioDataReader = zipArchiveReader.GetSubPath("audio");
         }
 
         public void PlaySoundEffectByName(string soundName) {
-            if (!_loadedSoundEffects.ContainsKey(soundName)) {
-                _loadedSoundEffects.TryAdd(soundName, Blish_HUD.BlishHud.ActiveContentManager.Load<SoundEffect>($"{soundName}"));
-            }
+            const string SOUND_EFFECT_FILE_EXTENSION = ".wav";
+            var filePath = soundName + SOUND_EFFECT_FILE_EXTENSION;
+            if (_audioDataReader.FileExists(filePath)) {
+                var soundEffect = _audioDataReader.GetFileStream(filePath);
+                var waveSource = new WaveFileReader(soundEffect);
+                
+                var soundOutput = new WasapiOut() { Device = GameService.GameIntegration.Audio.AudioDevice, };
+                soundOutput.Initialize(waveSource);
+                soundOutput.Volume = GameService.GameIntegration.Audio.Volume;
+
+                Task.Run(() => {
+                    soundOutput.Play();
+                    soundOutput.WaitForStopped();
+
+                    // This will dispose the sound effect stream and the wave source too
+                    soundOutput.Dispose();
+                });
```

Comment:
Because the object in `soundOutput` is instanciated in another Task than the Task.Run-Task, the method could return before that task reaches it's end. Therefore could it be disposed when the Task.Run wants to access the object. Probably the creation of the soundOutput should also be in that task so the ownership won't shift (and the `using`-Keyword can be used)

Reply:
I believe everything has been addressed elsewhere.  Were there any other changes needed here?